src/shared/blockdev-util.c

```diff
@@ -240,6 +240,52 @@ int lock_whole_block_device(dev_t devt, int operation) {
         return TAKE_FD(lock_fd);
 }
 
+int lookup_block_device(const char *p, dev_t *ret) {
+        dev_t rdev, dev = 0;
+        mode_t mode;
+        int r;
+
+        assert(p);
+        assert(ret);
+
+        r = device_path_parse_major_minor(p, &mode, &rdev);
+        if (r == -ENODEV) { /* not a parsable device node, need to go to disk */
+                struct stat st;
+
+                if (stat(p, &st) < 0)
+                        return log_warning_errno(errno, "Couldn't stat device '%s': %m", p);
```

Comment:
so we have this rule that "library-like" functions shouldn't log on their own, but let the caller do that. See https://systemd.io/CODING_STYLE/

You moved this into src/shared/, which made it a library-like function. (I mean, the rule isn't written in stone, it's OK to add functions that log on their own too, but usually they should be called `xyz_or_warn()` or so, to indicate the caller doesn't have to log anymore...